kore/src/Kore/Step/Simplification/Overloading.hs

```diff
@@ -465,7 +465,7 @@ mkInj ::
     Inj () ->
     TermLike RewritingVariableName ->
     TermLike RewritingVariableName
-mkInj inj injChild = (synthesize . InjF) inj{injChild}
+mkInj inj injChild = (synthesize . InjF) (inj :: Inj ()){injChild}
```

Comment:
Why is this necessary? The type of `inj` should be fixed by the function's signature.